Notify on leave if app is started

w-gao opened the pull request:
Problem
=======
On leave of page, notify the user before they leave the page if there are any unsaved changes [Jira](https://cruzhacks-dev.atlassian.net/browse/C2D-168)


Solution
========
What I/we did to solve this problem
* Add a `newChanges` ref to track if there are unsaved changes
  * useState doesn't work well with the unload event, but useRef seems to work


Change Summary:
---------------
* Notify if there are unsaved changes [5dd6af0]

Dev-Ops
=======
If you added an ENV variable, please check off that you've updated the following  
- [ ] Key & Sample value to `.env` & `sample.env` 
- [ ] GCP Secret Manager (Both development and production)
- [ ] Github Secrets (Added a development and production variable)
- [ ] Github Workflows `.github/workflows/dev.yml` & `.github/workflows/prod.yml`
- [ ] webpack-config.js

Images/Important Notes (optional):
-----------------------
* Insert any notes/issues, dependencies added or removed, or images

w-gao committed:
Notify if there are unsaved changes

krishnakpandian commented on src/views/Portal/components/ApplicationForm/ShortAnswer/index.view.tsx:
```diff
@@ -13,12 +14,13 @@ const ShortAnswerPage: React.FC = () => {
                   whyCruzHacks: e.target.value,
                 }))
-              }
+                setNewChanges()
+              }}
```
I think functionally it works but, it's a little inconsistent from my testing. It seems like this will only run onChange of the textContainer used. I think `setNewChanges` would need to be set on every input component.

w-gao replied:
Ooops, that’s my bad. I thought this was the `handleChange` function like the other ones. Good catch!

krishnakpandian reviewed (changes requested):
I think it's functioning how we would want it but the event listener doesn't seem to be run all the time with changes. My guess is, that the saveNewChanges() needs to be applied to every component

w-gao committed:
Fixes

w-gao commented:
I made some more changes since the last review. Primarily, I made it so it'll display the confirmation popup when new changes are made, even if they are "saved", so user won't accidentally close the tab and lose everything in session storage.

krishnakpandian reviewed (approved):
LGTM

w-gao committed:
Merge branch 'development' into feature/app-notify-unsaved-changes
